jib-core/src/main/java/com/google/cloud/tools/jib/tar/TarExtractor.java

```diff
@@ -72,6 +76,18 @@ public static void extract(Path source, Path destination) throws IOException {
           }
         }
       }
+      preserveModificationTimes(destination, entries);
+    }
+  }
+
+  private static void preserveModificationTimes(Path destination, List<TarArchiveEntry> entries)
+      throws IOException {
+    new DirectoryWalker(destination)
+        .filter(path -> Files.isDirectory(path))
+        .walk(path -> Files.setLastModifiedTime(path, FileTime.from(Instant.ofEpochSecond(1L))));
```

Comment:
As this is an unconventional behavior, I think we should apply this only when enabled. `TarExtractor` and `ZipUtil` are deemed general and isolated.

And maybe we can create an instance of `FileTime.fromMillis(1)` (I think it's equivalent) at the start of the method and avoid re-creating the instance everytime.

Reply:
That is a very good point. Done.

And a quick note as per our discussion offline: It looks like `DirectoryWalker.walk()` doesn't follow symbolic links. It calls `Files.walk(path)` which doesn't follow symbolic links (by default) unless `FileVisitOption.FOLLOW_LINKS` is passed as an argument.